codegen/main.rs

```diff
@@ -12,7 +12,7 @@ fn main() {
     Thread::spawn(move || {
         let output_dir = Path::new(os::getenv("OUT_DIR").unwrap());
         read_method::generate(output_dir).unwrap();
-    }).detach();
+    });
```

Comment:
Yes, right now this will totally detach the thread and can leave things undone when main exits and all threads are killed. Either we should use `let join = Thread::scope(..)` and let `join` drop at the end of main, or we should just drop the threading altogether.